packages/auth-core-plugins/src/AuthPluginParent.test.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { act, render, screen } from '@testing-library/react';
+import { dh } from '@deephaven/jsapi-shim';
+import { LoginOptions } from '@deephaven/jsapi-types';
+import AuthPluginParent from './AuthPluginParent';
+
+let mockParentResponse: Promise<LoginOptions>;
+jest.mock('@deephaven/jsapi-utils', () => ({
+  LOGIN_OPTIONS_REQUEST: 'mock-login-options-request',
+  requestParentResponse: jest.fn(() => mockParentResponse),
+}));
+
+function makeCoreClient() {
+  return new dh.CoreClient('wss://test.mockurl.example.com');
+}
+
+describe('availability tests', () => {
+  const client = makeCoreClient();
+  const authHandlers = [];
+  const authConfigValues = new Map();
+  it('is available when window opener is set', () => {
+    window.opener = { postMessage: jest.fn() };
+    window.history.pushState(
+      {},
+      'Test Title',
+      `/test.html?authProvider=parent`
+    );
+    expect(
+      AuthPluginParent.isAvailable(client, authHandlers, authConfigValues)
+    ).toBe(true);
+  });
+  it('is not available when window opener not set', () => {
+    delete window.opener;
+    expect(
+      AuthPluginParent.isAvailable(client, authHandlers, authConfigValues)
+    ).toBe(false);
+  });
+});
+
+function expectLoading() {
+  expect(screen.queryByTestId('auth-parent-loading')).not.toBeNull();
+}
+
+describe('component tests', () => {
+  const authConfigValues = new Map();
+
+  it('logs in when parent window provides login credentials', async () => {
+    let mockResolve;
+    mockParentResponse = new Promise(resolve => {
+      mockResolve = resolve;
+    });
+    const loginOptions = { token: 'mockParentToken' };
+    const loginPromise = Promise.resolve();
+    const onSuccess = jest.fn();
+    const onFailure = jest.fn();
+    const mockLogin = jest.fn(() => loginPromise);
+    const client = makeCoreClient();
+    client.login = mockLogin;
+    render(
+      <AuthPluginParent.Component
+        authConfigValues={authConfigValues}
+        client={client}
+        onFailure={onFailure}
+        onSuccess={onSuccess}
+      />
+    );
+    expectLoading();
+    expect(onSuccess).not.toHaveBeenCalled();
+    expect(onFailure).not.toHaveBeenCalled();
+    expect(mockLogin).not.toHaveBeenCalled();
+
+    // Send a message that should be ignored
+
+    expect(mockLogin).not.toHaveBeenCalled();
+    expect(onSuccess).not.toHaveBeenCalled();
```

Comment:
No message is sent here? And should also expect mockFailure to not be called again?